### ISSUE TITLE
poetry can't install, need another method

### ISSUE THREAD

**Issue by GalenMarek14** [imartinez/privateGPT#1119]
I tried to install this using my old username and it gave an error related to the username folder name so I changed it. Now error below keeps coming. I tried everything including uninstalling all of poetry and pyenv and freshly installing them but it just doesn't go away. No other Python installation gives this error. As you can see, the folder location (and I also set the global version as 3.11.6 too) properly sees 3.11 as the python version with its new location (OzkanPC) but poetry stubbornly tries to access the old one. I tried freshly installing all of them and even deleted the old privateGPT folder but it's still like this:

> D:\PrivateGPT\privateGPT>pyenv local 3.11.6
> 
> D:\PrivateGPT\privateGPT>python -V
> Python 3.11.6
> 
> D:\PrivateGPT\privateGPT>poetry install --with ui,local
> 
> Command ['C:\\Users\\OzkanPC\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\private-gpt-lGkbuM40-py3.11\\Scripts\\python.exe', '-I', '-W', 'ignore', '-'] errored with the following return code 103
> 
> Error output:
> No Python at '"C:\Users\Özkan PC\.pyenv\pyenv-win\versions\3.11.0\python.exe'
> 
> 
> Input:
> import sys
> 
> if hasattr(sys, "real_prefix"):
>     print(sys.real_prefix)
> elif hasattr(sys, "base_prefix"):
>     print(sys.base_prefix)
> else:
>     print(sys.prefix)
> 
> 
> D:\PrivateGPT\privateGPT>pyenv local 3.11.0
> 
> D:\PrivateGPT\privateGPT>python -V
> Python 3.11.0
> 
> D:\PrivateGPT\privateGPT>poetry install --with ui,local
> 
> Command ['C:\\Users\\OzkanPC\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\private-gpt-lGkbuM40-py3.11\\Scripts\\python.exe', '-I', '-W', 'ignore', '-'] errored with the following return code 103
> 
> Error output:
> No Python at '"C:\Users\Özkan PC\.pyenv\pyenv-win\versions\3.11.0\python.exe'
> 
> 
> Input:
> import sys
> 
> if hasattr(sys, "real_prefix"):
>     print(sys.real_prefix)
> elif hasattr(sys, "base_prefix"):
>     print(sys.base_prefix)
> else:
>     print(sys.prefix)

![image](https://github.com/imartinez/privateGPT/assets/148660654/452cffa2-4cd8-4bd7-8fdd-fdb40662a0ec)

Is there an alternate installation method? A manual one?

**Comment by KarlTheforest** [comment]
I recommend you **using vscode and create virtual environment** from there. I do once try to install it into my powershell. It give me almost problems the same as yours. By doing it into virtual environment, you can make the clean install. See tutorial on youtube on how to do so. cheers

**Comment by Jawn78** [comment]
@GalenMarek14  It looks like poetry is not finding python. What happens when you check your python & Poetry versions in the terminal

**Comment by GalenMarek14** [comment]
Sorry for the late reply. 
@KarlTheforest thank you. I have yet to try vscode but trying it with just using a normal virtual environment fixed all of the first problems. Now, I got another error at,

>PGPT_PROFILES=local make run

step. I will try the VSCode method, it will probably solve the entire thing. Thanks.

@Jawn78 the log I posted above answers the python question, it sees python version 3.11.6 in that folder. I made both local and global python versions the same with pyenv to be double sure.

As for poetry, I forgot to include that in the log but the command window sees that as well. So I don't know what causes this. But using a virtual environment solved poetry problems at least. Thanks.
